feat(blocky): add blocky (#3735)
truecharts/charts: charts/incubator/blocky/docs/installation-notes.md

# Installation notes

Blocky supports both single config file and multiple (split config files).
If you want to use multiple config files, set `useConfigDirectory` to true.

This will copy a basic config file (named `tc-config.yaml`) with redis and prometheus, and a default upstream
DNS of `1.1.1.1`. You can edit this file afterwards. If you ever need to recreate that file, simple delete it
and restart the Chart.

Scale users can change the `config` persistence to `HostPath (Simple)` and mount it to a directory on host.

> There you can add you config files.

Reviewer: Honestly, I don't want this.

The unique sellingpoint of blocky is that it can be setup compeletely stateless using configmaps.

# Installation notes

## Configuration

> This app pre-defines the following configuration:

Reviewer: ```suggestion
This app automatically includes a config file that pre-defines the following configuration:
```

> Redis (always):

Reviewer: ```suggestion
Redis (always present):
```

```yaml
redis:
  address: $redis_host:6379
  password: $redis_pass
  database: 0
  required: true
  connectionAttempts: 10
  connectionCooldown: 3s
```

> Prometheus (Only if enabled):

Reviewer: ```suggestion
Prometheus (Only present if enabled):
```

```yaml
prometheus:
  enable: true
  path: /metrics
```

> Whitelist/Blacklist (Only if hostPath or on values.yaml is defined):

Reviewer: ```suggestion
Whitelist/Blacklist (contains a file name if filepaths are defined in values.yaml and/or inline list of settings when the whitelist/blacklist are added to values.yaml) :
```

```yaml
blocking:
  whiteLists:
    ads:
      - whitelist.txt
      - |
        # inline definition with YAML literal block scalar style
  blackLists:
    ads:
      - blackist.txt
      - |
        # inline definition with YAML literal block scalar style
```

> ## TrueNAS Scale
> 
> - `Config File Host Path`: Define your host config file path
> - `Whitelist File Host Path`: Define your host whitelist file path
> - `Blacklist File Host Path`: Define your host whitelist file path
> 
> ## Helm Native
> 
> - Add your config in `values.yaml` under `blockyConfig:`
> - Add your whitelist in `values.yaml` under `blockyWhitelist:`
> - Add your blacklist in `values.yaml` under `blockyBlacklist:`

Reviewer: ```suggestion
## Configuration Instructions

- For native-helm, we advice to enter all configuration in `values.yaml`, this ensures blocky can run truely stateless, including any configuration.

-  TrueNAS SCALE, sadly enough, does not allow big text fields in the GUI. For that reason we've included an option to enter paths on the Host system (TrueNAS SCALE itself) for the App to fetch it's configuration, whitelist and/or blocklist from. This means SCALE does not rollback and changes to the config file and blocklist/whitelist.

### TrueNAS Scale

- `Config File Host Path`: Define your host config file path
- `Whitelist File Host Path`: Define your host whitelist file path
- `Blacklist File Host Path`: Define your host whitelist file path

### Helm Native

- Add your config in `values.yaml` under `blockyConfig:`
- Add your whitelist in `values.yaml` under `blockyWhitelist:`
- Add your blacklist in `values.yaml` under `blockyBlacklist:`

```

# Installation notes

## Configuration

This app automatically includes a config file that pre-defines some default settings.
This file will always gets read and it's content always used, your own config will be added in a seperate, additional, config file.

The following config will be pre-configured:

Redis (always present):

```yaml
redis:
  address: $redis_host:6379
  password: $redis_pass
  database: 0
  required: true
  connectionAttempts: 10
  connectionCooldown: 3s
```

Prometheus (Only present if enabled):

```yaml
prometheus:
  enable: true
  path: /metrics
```

> Whitelist/Blacklist (contains a file name if filepaths are defined in values.yaml and/or inline list of settings when the whitelist/blacklist are added to values.yaml) :

Reviewer: ```suggestion
Whitelist/Blacklist (from values.yaml) :
```

> ```yaml
> blocking:
>   whiteLists:
>     ads:
>       - whitelist.txt
>       - # Content from .Values.blockyWhiteList
>   blackLists:
>     ads:
>       - blackist.txt
>       - # Content from .Values.blockyBlackList

Reviewer: ```suggestion
```yaml
blocking:
  whiteLists:
    ads:
      - # Content from .Values.blockyWhiteList
  blackLists:
    ads:
      - # Content from .Values.blockyBlackList
```

## Configuration Instructions

> - We offer botht he option of adding Config/Whitelist/Blacklist by directly mounting a configfile or by adding the configuration/list content by adding it to values.yaml. Both options can be used at the same time as you see fit
> 
> 
> 
> ### Options
> 
> - `Config File Host Path`: Define your host config file path
> - `Whitelist File Host Path`: Define your host whitelist file path
> - `Blacklist File Host Path`: Define your host whitelist file path

Reviewer: ```suggestion
```

> - Add your config in `values.yaml` under `blockyConfig:`
> - Add your whitelists in `values.yaml` under `blockyWhitelist:`
> - Add your blacklists in `values.yaml` under `blockyBlacklist:`
> 

Reviewer: ```suggestion
- Add your config in `values.yaml` under `blockyConfig:`
- Add your whitelists in `values.yaml` under `blockyWhitelist:`
- Add your blacklists in `values.yaml` under `blockyBlacklist:`

### Adding config by mounting files

You can mount custom config files, using `persistence` or, in SCALE GUI, `Additional Storage` to the following path:
`/app/configpath`

You can mount custom Whitelist/Blacklist files, using `persistence` or, in SCALE GUI, `Additional Storage` and enter the path in your whitelist or blacklist settings manually
```

# Installation notes

## Configuration

> This app automatically includes a config file that pre-defines the following configuration:

Reviewer: ```suggestion
This app automatically includes a config file that pre-defines some default settings.
This file will always gets read and it's content always used, your own config will be added in a seperate, additional, config file.

The following config will be pre-configured:
```

Redis (always present):

```yaml
redis:
  address: $redis_host:6379
  password: $redis_pass
  database: 0
  required: true
  connectionAttempts: 10
  connectionCooldown: 3s
```

Prometheus (Only present if enabled):

```yaml
prometheus:
  enable: true
  path: /metrics
```

Whitelist/Blacklist (contains a file name if filepaths are defined in values.yaml and/or inline list of settings when the whitelist/blacklist are added to values.yaml) :

```yaml
blocking:
  whiteLists:
    ads:
      - whitelist.txt
      - |
        # inline definition with YAML literal block scalar style
  blackLists:
    ads:
      - blackist.txt
      - |
        # inline definition with YAML literal block scalar style
```

## Configuration Instructions

> 
> - For native-helm, we advice to enter all configuration in `values.yaml`, this ensures blocky can run truely stateless, including any configuration.
> 
> -  TrueNAS SCALE, sadly enough, does not allow big text fields in the GUI. For that reason we've included an option to enter paths on the Host system (TrueNAS SCALE itself) for the App to fetch it's configuration, whitelist and/or blocklist from. This means SCALE does not rollback and changes to the config file and blocklist/whitelist.
> 

Reviewer: ```suggestion

- For native-helm, we advice to enter all configuration in `values.yaml`, this ensures blocky can run truely stateless, including any configuration.

- TrueNAS SCALE, sadly enough, does not allow big text fields in the GUI. For that reason we've included an option to enter paths on the Host system (TrueNAS SCALE itself) for the App to fetch it's configuration, whitelist and/or blocklist from. This means SCALE does not rollback and changes to the config file and blocklist/whitelist.

```

# Installation notes

## Configuration

This app automatically includes a config file that pre-defines some default settings.
This file will always gets read and it's content always used, your own config will be added in a seperate, additional, config file.

The following config will be pre-configured:

Redis (always present):

```yaml
redis:
  address: $redis_host:6379
  password: $redis_pass
  database: 0
  required: true
  connectionAttempts: 10
  connectionCooldown: 3s
```

Prometheus (Only present if enabled):

```yaml
prometheus:
  enable: true
  path: /metrics
```

Whitelist/Blacklist (contains a file name if filepaths are defined in values.yaml and/or inline list of settings when the whitelist/blacklist are added to values.yaml) :

```yaml
blocking:
  whiteLists:
    ads:
      - whitelist.txt
      - |
        # inline definition with YAML literal block scalar style
  blackLists:
    ads:
      - blackist.txt
      - |
        # inline definition with YAML literal block scalar style

> ```

Reviewer: ```suggestion
```yaml
blocking:
  whiteLists:
    ads:
      - whitelist.txt
      - # Content from .Values.blockyWhiteList
  blackLists:
    ads:
      - blackist.txt
      - # Content from .Values.blockyBlackList
```
```

> ## Configuration Instructions
> 
> - For native-helm, we advice to enter all configuration in `values.yaml`, this ensures blocky can run truely stateless, including any configuration.
> 
> - TrueNAS SCALE, sadly enough, does not allow big text fields in the GUI. For that reason we've included an option to enter paths on the Host system (TrueNAS SCALE itself) for the App to fetch it's configuration, whitelist and/or blocklist from. This means SCALE does not rollback and changes to the config file and blocklist/whitelist.
> 
> 
> ### TrueNAS Scale
> 
> - `Config File Host Path`: Define your host config file path
> - `Whitelist File Host Path`: Define your host whitelist file path
> - `Blacklist File Host Path`: Define your host whitelist file path
> 
> ### Helm Native
> 
> - Add your config in `values.yaml` under `blockyConfig:`
> - Add your whitelist in `values.yaml` under `blockyWhitelist:`
> - Add your blacklist in `values.yaml` under `blockyBlacklist:`

Reviewer: ```suggestion
## Configuration Instructions

- We offer botht he option of adding Config/Whitelist/Blacklist by directly mounting a configfile or by adding the configuration/list content by adding it to values.yaml. Both options can be used at the same time as you see fit

### Options

- `Config File Host Path`: Define your host config file path
- `Whitelist File Host Path`: Define your host whitelist file path
- `Blacklist File Host Path`: Define your host whitelist file path

- Add your config in `values.yaml` under `blockyConfig:`
- Add your whitelists in `values.yaml` under `blockyWhitelist:`
- Add your blacklists in `values.yaml` under `blockyBlacklist:`
```
